### PR TITLE
KAFKA-17529: Remove blacklist from MM2

### PR DESCRIPTION
**Pull request by chiacyu**
As titled, check [KAFKA-17529](https://issues.apache.org/jira/browse/KAFKA-17529) for details. 
Please take a look, thanks!

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@chiacyu thanks for your patch. Could you please cleanup the test of https://github.com/apache/kafka/blob/c95865437b80ad37cc2bf3007fc744b40f712aa8/connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceConfigTest.java#L70

Also, please update `upgrade.html`

File: docs/upgrade.html [hunk 1]
```diff
@@ -51,6 +51,18 @@ <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4
                 The <code>--whitelist</code> option was removed from the <code>kafka-console-consumer</code> command line tool.
                 Please use <code>--include</code> instead.
             </li>
+            <li>
+                The <code>--whitelist</code> option was removed from the <code>kafka-console-consumer</code> command line tool.
```
Comment (chia7712) [inline docs/upgrade.html]: Please remove this unrelated change

**Review by chia7712 — COMMENTED** [review]
@chiacyu thanks for your patch. please check `testConfigBackwardsCompatibility` also

File: connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceConfigTest.java [hunk 2]
```diff
@@ -69,8 +69,8 @@ public void testConfigPropertyMatching() {
     @Test
     public void testConfigBackwardsCompatibility() {
```
Comment (chia7712) [inline connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceConfigTest.java]: we don't need this test because the `blacklist`-related configs are removed, right?

**Review by chia7712 — COMMENTED** [review]
@chiacyu thanks for updated patch. Could you please remove `groups.blacklist` also?

File: docs/upgrade.html [hunk 3]
```diff
@@ -51,6 +51,18 @@ <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4
                 The <code>--whitelist</code> option was removed from the <code>kafka-console-consumer</code> command line tool.
                 Please use <code>--include</code> instead.
             </li>
+            <li>
+                The <code>config.properties.blacklist</code> was removed from the <code>kafka.connect.mirror.MirrorSourceConfig</code>
+                Please use <code>config.properties.exclude</code> instead.
+            </li>
+            <li>
+                The <code>topics.blacklist</code> was removed from the <code>kafka.connect.mirror.MirrorSourceConfig</code>
```
Comment (chia7712) [inline docs/upgrade.html]: `org.apache.kafka.connect.mirror.MirrorSourceConfig`

File: docs/upgrade.html [hunk 4]
```diff
@@ -51,6 +51,18 @@ <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4
...
+                Please use <code>topics.exclude</code> instead.
+            </li>
+            <li>
+                The <code>groups.blacklist</code> was removed from the <code>kafka.connect.mirror.MirrorSourceConfig</code>
```
Comment (chia7712) [inline docs/upgrade.html]: `org.apache.kafka.connect.mirror.MirrorSourceConfig`

File: docs/upgrade.html [hunk 5]
```diff
@@ -51,6 +51,18 @@ <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4
                 The <code>--whitelist</code> option was removed from the <code>kafka-console-consumer</code> command line tool.
                 Please use <code>--include</code> instead.
             </li>
+            <li>
+                The <code>config.properties.blacklist</code> was removed from the <code>kafka.connect.mirror.MirrorSourceConfig</code>
```
Comment (chia7712) [inline docs/upgrade.html]: `org.apache.kafka.connect.mirror.MirrorSourceConfig`

**Review by chia7712 — COMMENTED** [review]
@chiacyu thanks for your updates

File: connect/mirror/src/main/java/org/apache/kafka/connect/mirror/DefaultGroupFilter.java [hunk 6]
```diff
@@ -73,16 +72,11 @@ static class GroupFilterConfig extends AbstractConfig {
...
-                    GROUPS_EXCLUDE_DOC)
-            .define(GROUPS_EXCLUDE_CONFIG_ALIAS,
-                    Type.LIST,
-                    null,
-                    Importance.HIGH,
-                    "Deprecated. Use " + GROUPS_EXCLUDE_CONFIG + " instead.");
+                    GROUPS_EXCLUDE_DOC);
 
         GroupFilterConfig(Map<String, ?> props) {
             super(DEF, ConfigUtils.translateDeprecatedConfigs(props, new String[][]{
```
Comment (chia7712) [inline connect/mirror/src/main/java/org/apache/kafka/connect/mirror/DefaultGroupFilter.java]: we don't need to call `translateDeprecatedConfigs` now, since the deprecated configs are removed

File: connect/mirror/src/main/java/org/apache/kafka/connect/mirror/DefaultConfigPropertyFilter.java [hunk 7]
```diff
@@ -89,7 +83,7 @@ static class ConfigPropertyFilterConfig extends AbstractConfig {
 
         ConfigPropertyFilterConfig(Map<String, ?> props) {
             super(DEF, ConfigUtils.translateDeprecatedConfigs(props, new String[][]{
```
Comment (chia7712) [inline connect/mirror/src/main/java/org/apache/kafka/connect/mirror/DefaultConfigPropertyFilter.java]: we don't need to call `translateDeprecatedConfigs` now, since the deprecated configs are removed

File: connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConfig.java [hunk 8]
```diff
@@ -81,7 +80,7 @@ public class MirrorCheckpointConfig extends MirrorConnectorConfig {
 
     public MirrorCheckpointConfig(Map<String, String> props) {
         super(CONNECTOR_CONFIG_DEF, ConfigUtils.translateDeprecatedConfigs(props, new String[][]{
```
Comment (chia7712) [inline connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConfig.java]: we don't need to call `translateDeprecatedConfigs` now, since the deprecated configs are removed

**Review by chia7712 — COMMENTED** [review]
@chiacyu thanks for updates

File: connect/mirror/src/main/java/org/apache/kafka/connect/mirror/DefaultTopicFilter.java [hunk 9]
```diff
@@ -72,16 +71,11 @@ static class TopicFilterConfig extends AbstractConfig {
...
-                    TOPICS_EXCLUDE_DOC)
-            .define(TOPICS_EXCLUDE_CONFIG_ALIAS,
-                    Type.LIST,
-                    null,
-                    Importance.HIGH,
-                    "Deprecated. Use " + TOPICS_EXCLUDE_CONFIG + " instead.");
+                    TOPICS_EXCLUDE_DOC);
 
         TopicFilterConfig(Map<String, ?> props) {
             super(DEF, ConfigUtils.translateDeprecatedConfigs(props, new String[][]{
```
Comment (chia7712) [inline connect/mirror/src/main/java/org/apache/kafka/connect/mirror/DefaultTopicFilter.java]: it does not need `translateDeprecatedConfigs`, right?

**Review by TaiJuWu — APPROVED** [review]

File: docs/upgrade.html [hunk 10]
```diff
@@ -66,6 +75,8 @@ <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4
                         <li>The <code>--whitelist</code> option was removed from the <code>kafka-console-consumer</code> command line tool.
                             Please use <code>--include</code> instead.
                         </li>
+                        <li>The <code>--delete-config</code> option in the <code>kafka-topics</code> command line tool has been deprecated.
```
Comment (chia7712) [inline docs/upgrade.html]: Please remove the unnecessary changes

**Review by chia7712 — APPROVED** [review]

File: connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConfig.java [hunk 11]
```diff
@@ -98,8 +96,8 @@ public class MirrorSourceConfig extends MirrorConnectorConfig {
 
     public MirrorSourceConfig(Map<String, String> props) {
         super(CONNECTOR_CONFIG_DEF, ConfigUtils.translateDeprecatedConfigs(props, new String[][]{
```
Comment (chia7712) [inline connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConfig.java]: we don't need `translateDeprecatedConfigs`, right?
